src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -92,10 +92,11 @@ class MessageBufferProcessor(
 		val batchSizeWithExtraGap = batchSize * 2
 		val messages = messageStore.loadLockedMessages(batchSizeWithExtraGap)
 		scala.util.Random.shuffle(producers).foreach {
-			case (exchange, producer) =>
-				val messagesToProducer = messages.filter(_.exchangeName == producer.exchange.name)
+			case (exchange: String, producer: AmqpProducerInterface) =>
```

Comment:
Similarly this code

```scala
class Test[A](test: Option[A]) {

	test match {
		case Some(a) => "case1"
		case _ => "case2"
	}
}
```

translates to

```scala
package <empty> {
  class Test extends Object {
    <paramaccessor> private[this] val test: Option = _;
    def <init>(test: Option): Test = {
      Test.super.<init>();
      ()
    };
    {
      case <synthetic> val x1: Option = Test.this.test;
      case6(){
        if (x1.$isInstanceOf[Some]())
          {
            <synthetic> val x2: Some = (x1.$asInstanceOf[Some](): Some);
            {
              val a: Object = x2.x();
              if (a.$isInstanceOf[Object]())
                matchEnd5("case1")
              else
                case7()
            }
          }
        else
          case7()
      };
      case7(){
        matchEnd5("case2")
      };
      matchEnd5(x: String){
        x
      }
    }
  }
}
```

and you even get an erasure warning because of `isInstanceOf[Object]`.